src/components/Auth/Auth.js

```diff
@@ -0,0 +1,71 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Redirect } from 'react-router-dom';
+import { Form, Field } from 'react-final-form';
+
+class Auth extends Component {
+  componentDidMount() {
+    return (this.props.register.isLoggedIn ? <Redirect to="/home" /> : '');
+  }
+  onSignup = (formData) => {
+    // console.log(formData);
+    this.props.registerUser(formData);
+  }
+  render() {
+    return (
+      <div>
+        <Form
+          onSubmit={this.onSignup}
+          render={({
+            handleSubmit, form, pristine,
+          }) => (
+            <form onSubmit={handleSubmit}>
+              <div>
+                <Field
+                  name="email"
+                  component="input"
+                  type="email"
+                  placeholder="Enter your email"
+                  id="email"
+                  required
+                />
+              </div>
+              <div>
+                <Field
+                  name="password"
+                  component="input"
+                  type="password"
+                  placeholder="Enter your password"
+                  required
+                />
+              </div>
+              <div className="buttons">
+                <button type="submit" disabled={pristine}>
```

Comment:
The button should be disabled on `pristine` and `submitErrors` or other field-level validation checks.

Reply:
Done!